### PR TITLE
Unit Test for k8s/Users

### PR DESCRIPTION
**Pull request by ashley-o0o**
Closes: [RHOAIENG-2175](https://issues.redhat.com/browse/RHOAIENG-2175)

## Description
Unit test case for: frontend/src/api/k8s/users.ts

## How Has This Been Tested?
'npm run test'

## Test Impact
Coverage:
Screenshot 2024-02-07 at 2 57 01 PM

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by pnaik1** [comment]
@ashley-o0o  the  test looks good but can you please squash the commits?

File: frontend/src/api/k8s/__tests__/users.spec.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  },
+  groups: [],
+};
+describe('listUser', () => {
+  it('should list users without label selector', async () => {
+    k8sListResourceMock.mockResolvedValue(mockK8sResourceList([userMock]));
+    const result = await listUsers();
+    expect(k8sListResourceMock).toHaveBeenCalledTimes(1);
+    expect(k8sListResourceMock).toHaveBeenCalledWith({
+      model: { apiGroup: 'user.openshift.io', apiVersion: 'v1', kind: 'User', plural: 'users' },
```
Comment (uidoyen) [inline frontend/src/api/k8s/__tests__/users.spec.ts:25]: ` model: UserModel` can be used instead.

**Review by uidoyen — CHANGES_REQUESTED** [review]

File: frontend/src/api/k8s/__tests__/users.spec.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+const userMock: UserKind = {
+  apiVersion: 'v1',
+  kind: 'userkind',
+  metadata: {
+    name: 'name',
+    namespace: 'namespace',
+  },
+  groups: [],
+};
+describe('listUser', () => {
```
Comment (ppadti) [inline frontend/src/api/k8s/__tests__/users.spec.ts:20]: typo:  `listUser `-> `listUsers`

**Review by uidoyen — APPROVED** [review]

**Pull request merged**